core/store/src/trie/mod.rs

```diff
@@ -757,6 +766,97 @@ impl Trie {
     pub fn get_trie_nodes_count(&self) -> TrieNodesCount {
         self.storage.get_trie_nodes_count()
     }
+
+    pub fn get_proof(
+        &self,
+        root: &CryptoHash,
+        key: &[u8],
+    ) -> Result<Option<Vec<Level>>, StorageError> {
```

Comment:
slightly off-topic.  Wouldn't `Result<Vec<Level>, StorageError>` suffice here?  Isn't a vector of length 0 sufficient to capture the `None` situation or do we need to represent them differently?

Reply:
We can default to that.